components/courses/app/controllers/admin/courses/mentors_controller.rb

```diff
@@ -18,15 +19,12 @@ def create
       end
 
       def destroy
-        react_to mentor.destroy
+        return react_to mentor.destroy unless mentor == current_season.mentors.first
```

Comment:
business logic in controller detected(!)

Please, create policy -> `Courses::Mentor::DestroyPolicy` and it should decide this.